### ISSUE TITLE
sparql response is empty

### ISSUE THREAD

**Issue by mmisw** [mmisw/mmiorr#1]
_From [caru...@gmail.com](https://code.google.com/u/113886747689301365533/) on October 16, 2008 21:43:00_

What steps will reproduce the problem? 1. Open http://mmisw.org/ont_tests/sparql.html and write the query:
  
PREFIX rdfs:  \< http://www.w3.org/2000/01/rdf-schema# >
 CONSTRUCT  { ?s ?p ?o } where{?s rdfs:label ?o. } LIMIT 20

2. Summit query

3. Response is:
\<rdf:RDF
   xmlns:rdf=" http://www.w3.org/1999/02/22-rdf-syntax-ns# "
   xmlns:rdfs=" http://www.w3.org/2000/01/rdf-schema# " >
\</rdf:RDF> What is the expected output? What do you see instead? Expected output should be non-empty.

Note that the query:
   CONSTRUCT  { ?s ?p ?o } where{?s ?p ?o. } LIMIT 20
works ok. So, the problems seems related with URIs being included in the query.

I already confirmed that there is no encoding/decoding problems: the server
gets exactly what is submitted.

So, the problem should be in the way the query is processed. But I'm
following the same logic as with the grails-based registry (which
presumably works ok).

Here is the main source: https://code.google.com/p/mmisw/source/browse/trunk/org.mmisw.ont/src/org/mmisw/ont/OntGraph.java See the getRDF(String sparqlQuery) method, which is the java version
of the same groovy method in: https://code.google.com/p/remembermyid/source/browse/trunk/Registry/grails-app/services/SparqlService.groovy

_Original issue: http://code.google.com/p/mmisw/issues/detail?id=1_

**Comment by mmisw** [comment]
_From [caru...@gmail.com](https://code.google.com/u/113886747689301365533/) on October 16, 2008 21:48:27_

**Labels:** org.mmisw.ont

**Comment by mmisw** [comment]
_From [caru...@gmail.com](https://code.google.com/u/113886747689301365533/) on October 17, 2008 13:03:22_

According to some tests against the regtest sparql service:

WSDL: http://mmisw.org/regtest/services/sparql?wsdl operation: getRDF

it seems the same problem happens in that version.  Luis: can you do your own tests
to confirm?

**Comment by mmisw** [comment]
_From [caru...@gmail.com](https://code.google.com/u/113886747689301365533/) on October 19, 2008 22:36:31_

Hmm, it seems the problem is simply that the queries are not correctly written.

The query reported here, when written as follows:

PREFIX rdfs:  \< http://www.w3.org/2000/01/rdf-schema# >
CONSTRUCT  { ?s rdfs:label ?o } 
WHERE {?s rdfs:label ?o. } LIMIT 20

just works fine. Note the rdfs:label instead of only ?p in the CONSTRUCT part.

(well, this is a good way to learn more about jena and arq ;-) )

I created the following page with more examples, mainly using the CF, device, and
platform ontologies (which are the ones registered at the time of writing): http://mmisw.org/ont/sparql.html Marking as fixed.

The "sparql" command also accepts SELECT besides CONSTRUCT statements.  

Note: I refactored the SPARQL related code although it's still considered "unfinished".

**Status:** Fixed
